src/Insolation.jl

```diff
@@ -1,47 +1,68 @@
-const ϖ_spline_etp = Ref(CubicSplineInterpolation(mock_t_range, mock_x_data, extrapolation_bc = NaN))
-function __init__()
-    datapath = joinpath(@__DIR__, "../src/data/", "INSOL.LA2004.BTL.csv");
-    x, _ = readdlm(datapath, ',', Float64, header=true);
-    t_range = x[1,1]*1e3 : 1e3 : x[end,1]*1e3; # array of every 1 kyr to range of years
-    e_spline_etp[] = CubicSplineInterpolation(t_range, x[:,2], extrapolation_bc = NaN);
-    γ_spline_etp[] = CubicSplineInterpolation(t_range, x[:,3], extrapolation_bc = NaN);
-    ϖ_spline_etp[] = CubicSplineInterpolation(t_range, x[:,4], extrapolation_bc = NaN);
-end
-function e_spline(args...)
-    return e_spline_etp[](args...)
-end
-function γ_spline(args...)
-    return γ_spline_etp[](args...)
-end
-function ϖ_spline(args...)
-    return ϖ_spline_etp[](args...)
+function orbital_parameters_dataset_path()
+    era_dataset = AW.ArtifactWrapper(
+        @__DIR__,
+        "era-global",
+        AW.ArtifactFile[AW.ArtifactFile(
+            url = "https://caltech.box.com/shared/static/3y02smnlxhgwednm3eho7lve2xhq1n7r.csv",
+            filename = "INSOL.LA2004.BTL.csv",
+        ),],
+    )
+    return AW.get_data_folder(era_dataset)
 end
 
 """
-    orbital_params(dt::FT) where {FT <: Real}
+    OrbitalData
 
-This function returns the orbital parameters (ϖ, γ, e) at dt (years) since J2000 epoch.
-Data are read from file and interpolation functions are created in __init__() method.
-The functions are stored as global variables that are used inside Insolation.jl.
 The parameters vary due to Milankovitch cycles. 
-Orbital parameters from the Laskar 2004 paper are in the "src/data/INSOL.LA2004.BTL.csv" file.
+
+Orbital parameters from the Laskar 2004 paper are
+lazily downloaded from Caltech Box to the
+`orbital_parameters_dataset_path()` path.
+"""
+struct OrbitalData{E, G, O}
+    e_spline_etp::E
+    γ_spline_etp::G
+    ϖ_spline_etp::O
+    function OrbitalData()
+        datapath = joinpath(orbital_parameters_dataset_path(), "INSOL.LA2004.BTL.csv");
+        x, _ = readdlm(datapath, ',', Float64, header=true);
+        t_range = x[1,1]*1e3 : 1e3 : x[end,1]*1e3; # array of every 1 kyr to range of years
+        e_spline_etp = CubicSplineInterpolation(t_range, x[:,2], extrapolation_bc = NaN);
+        γ_spline_etp = CubicSplineInterpolation(t_range, x[:,3], extrapolation_bc = NaN);
+        ϖ_spline_etp = CubicSplineInterpolation(t_range, x[:,4], extrapolation_bc = NaN);
+
+        E = typeof(e_spline_etp)
+        G = typeof(γ_spline_etp)
+        O = typeof(ϖ_spline_etp)
+        return new{E,G,O}(e_spline_etp,γ_spline_etp,ϖ_spline_etp)
+    end
+end
+
+e_spline(od, args...) = od.e_spline_etp(args...)
+γ_spline(od, args...) = od.γ_spline_etp(args...)
+ϖ_spline(od, args...) = od.ϖ_spline_etp(args...)
+
+"""
+    orbital_params(od::OrbitalData, dt::FT) where {FT <: Real}
+
+This function returns the orbital parameters (ϖ, γ, e)
+at dt (years) since J2000 epoch.
```

Comment:
Parameters are interpolated from the values given in the 
Laskar 2004 dataset using a cubic spline interpolation.